packages/sdk-snippets/src/helpers/code-writer.ts

```diff
@@ -0,0 +1,66 @@
+import CodeBlockWriter from 'code-block-writer';
+
+import { escapeForDoubleQuotes, escapeForSingleQuotes } from './escape';
+
+export class CodeWriter extends CodeBlockWriter {
+  private ranges: {
+    security?: Record<string, { line: number }>;
+    server?: Record<string, { line: number }>;
+  } = {};
+
+  private _texts: string[] = [];
```

Comment:
Because `_texts` is private in `CodeBlockWriter` we can't access it here to determine where in the snippet we are when we're adding a range, so I'm having to override it.

Reply:
This is unfortunate, but hey if it works!